### PR TITLE
feat: Add support for 'bun' package manager

### LINKED ISSUE #1078: Bun build support
## Feature Request
Thanks for your suggestion. Feel free to just state your idea. Writing the issue, it would better to filling the below items:

  - A description of the problem you're trying to solve.
  - An overview of the suggested solution.
  - Examples of how the suggestion would work in various places.
    - Code examples showing the expected behavior.

`randexp.js` and thier dependency `DRange` is complied by old CJS.
It works fine with node.js, but `bun build` fails bundling ( I know `typia` does not support bun officially)

Related: https://github.com/ryoppippi/unplugin-typia/issues/44

For solving, I'll implement an alternative of `randexp.js` 

(This is just suggestion. If you do not want it, close this issue)

### PR DESCRIPTION
**Pull request by ryoppippi**
fixes: #1078
This commit introduces support for the 'bun' package manager in the
TypiaSetupWizard and updates the documentation accordingly. 

If you created a new feature, then create the unit test function, too.

```bash
# COMPILE
npm run build

# RE-WRITE TEST PROGRAMS IF REQUIRED
npm run test:template

# BUILD TEST PROGRAM
npm run build:test

# DO TEST
npm run test
```

Learn more about the [CONTRIBUTING](CONTRIBUTING.md)

### CODE REVIEW

**Comment by ryoppippi** [comment]
Also I addedd the test with bun!

**Review by samchon — COMMENTED** [review]
As I'm using Windows laptop, I need to investigate about the `which` command.

**Comment by ryoppippi** [comment]
@samchon 
Then, I'll use [dax's which command](https://github.com/dsherret/dax#custom-cross-platform-shell-commands)

**Comment by samchon** [comment]
I don't know about bun a lot.

Does `build.yml` test both on NodeJS and Bun?

File: deploy/index.ts [hunk 1]
```diff
@@ -93,6 +99,11 @@ const main = (): void => {
       : ["npm run build", "npm start"],
   );
   test(version)("test-esm")(["npm run build", "npm start"]);
+  /* if bun is available, test with bun */
+  if (await detectComanndAvailable("bun")) {
+    test(version)("test")(["bun --bun run build_run", "bun --bun run start"]);
+    test(version)("test-esm")(["bun --bun run build", "bun --bun run start"]);
+  }
```
Comment (ryoppippi) [inline deploy/index.ts:106]: This is the command where bun executes the test

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,6 +20,9 @@ jobs:
       - uses: actions/setup-node@v4
         with:
           node-version: 20.x
+      - uses: oven-sh/setup-bun@v1
+        with:
+          bun-version: latest
```
Comment (ryoppippi) [inline .github/workflows/build.yml:25]: This is whrere GHA installs Bun

**Comment by ryoppippi** [comment]
Yes it runs on Bun!
Take a look at [The CI Result](https://github.com/samchon/typia/actions/runs/9430634474/job/25978384372?pr=1080)
You can search it with the term of `Bun`

**Comment by samchon** [comment]
Okay, good.

At last, would you fix the confliction of `setup.mdx` file as you want?

**Comment by ryoppippi** [comment]
@samchon I resolved conflicts!

**Comment by samchon** [comment]
Merged, but it would better to introduce on `unplugin-typia` section as an example.

Wait for your PR.

https://jsr.io/@ryoppippi/unplugin-typia/doc/bun/~/default

**Comment by ryoppippi** [comment]
@samchon OK!

**Pull request merged**